### PR TITLE
ICS loose ends

### PR DESCRIPTION
**Pull request by sampocs**
## Context and purpose of the change
This PR involves whitelisting missing param keys and a helper function for registering stToken denoms to consumer whitelist.
stTokens need to be whitelisted on consumer chain to be flown to provider chain and distributed to hub validators.
These updates were originally included into this [PR](https://github.com/Stride-Labs/stride/pull/860) but we pulled ICS updates from the PR for the convenience.

## Brief Changelog
  - *Implement a function for registering stTokens as consumer rewards denom to the whitelist*
  - *Add missing param keys to proposal whitelist*
  - *Register stTokens to consumer reward denom whitelist automatically when new hostzones are added*

## Author's Checklist
- [x] Run and PASSED locally all integration tests.

### CODE REVIEW

**Comment by asalzmann** [comment]
@strbrian can you add a more detailed description to this PR?

**Comment by strbrian** [comment]
> @strbrian can you add a more detailed description to this PR?

Updated the PR description.

File: app/proposals_whitelisting.go [hunk 1]
```diff
@@ -86,6 +88,11 @@ var WhitelistedParams = map[ccvgov.ParamChangeKey]struct{}{
 	{MsgType: stakingtypes.ModuleName, Key: string(stakingtypes.KeyMaxEntries)}:        {},
 	{MsgType: stakingtypes.ModuleName, Key: string(stakingtypes.KeyHistoricalEntries)}: {},
 	{MsgType: stakingtypes.ModuleName, Key: string(stakingtypes.KeyBondDenom)}:         {},
+	//autopilot
```
Comment (asalzmann) [inline app/proposals_whitelisting.go:91]: @strbrian just checking - did you go through all of our modules and make sure all of the param change proposals are included here?

File: x/stakeibc/keeper/consumer.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+// Register new stTokens to the consumer reward denom whitelist
+func (k Keeper) RegisterStTokenDenomsToWhitelist(ctx sdk.Context, denoms []string) error {
+	hostZones := k.GetAllHostZone(ctx)
+	allDenomsMap := make(map[string]bool)
+	registeredDenomsMap := make(map[string]bool)
+
+	// get all stToken denoms
+	for _, zone := range hostZones {
+		allDenomsMap["st"+zone.HostDenom] = true
```
Comment (asalzmann) [inline x/stakeibc/keeper/consumer.go:17]: please use `StAssetDenomFromHostZoneDenom`

https://github.com/Stride-Labs/stride/blob/dfc5fb0fb33a471f86f1f74992ff3809919e7e80/x/stakeibc/types/message_liquid_stake.go#L31

**Review by asalzmann — COMMENTED** [review]
Looks good - we also discussed registering new denoms when a new host zone is added.

How are you thinking this will work?

File: app/proposals_whitelisting.go (hunk 1 above)
Comment (strbrian) [reply app/proposals_whitelisting.go:93]: yes, I have gone through all of our modules and no missing params.
But there are still some missing params for sdk and ics consumer and I added them to the whitelist additionally.

File: x/stakeibc/keeper/consumer.go (hunk 2 above)
Comment (strbrian) [reply x/stakeibc/keeper/consumer.go]: ok

File: app/proposals_whitelisting.go (hunk 1 above)
Comment (asalzmann) [reply app/proposals_whitelisting.go:93]: Thanks! We'll just have to remember to always add new params here going forward

**Comment by strbrian** [comment]
> lgtm! when we start dockernet, stuatom should now be registered as a fee denom - let's verify that as we test the upgrade

exactly, I actually already verified that as well.
once the dockernet is up, we can simply query consumer params using this link and check reward denom array.
http://localhost:1317/interchain_security/ccv/consumer/params

**Pull request merged**